aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -294,6 +298,28 @@ inline Return Dispatcher::callWithDispatchKey(const OperatorHandle& op, Dispatch
   detail::unused_arg_(args...);  // workaround for a false-positive warning about unused parameters in gcc 5
   const auto& dispatchTable = op.operatorIterator_->op.dispatch_table();
   const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+  // Check if we need to run callbacks registered with RecordFunction
+  // If true and callbacks need inputs, we box the arguments and pass
+  // them into the callbacks and also into the kernel call
+
+  // Note: for perf reasons we wouldn't want to pass arguments into
+  // the function call or prematurely box them
+  at::RecordFunction guard(at::RecordScope::FUNCTION);
+  if (guard.active) {
+    if (guard.needs_inputs) {
+      std::vector<c10::IValue> stack;
+      auto boxed_all_args = impl::boxArgumentsIntoStack(stack, args...);
+
+      guard.before(op.schema().name(), stack, at::sequence_number::peek());
+
+      // if we could convert all the arguments, also pass the stack into the kernel call
+      if (boxed_all_args) {
+        return kernel.template callBoxedOrUnboxed<Return, Args...>(op, stack, std::forward<Args>(args)...);
```

Comment:
it's a trade off between code simplicity and a small win of not boxing twice; I'm mostly neutral about it here, but @smessmer might have something more to add and maybe give an answer regarding if callBoxedOrUnboxed going to be used later elsewhere

Reply:
Yeah this only exists to prevent having to box arguments twice. I think doing an extra boxing adds a small perf overhead that probably doesn't matter much and adding this `callBoxedOrUnboxed` helper adds a small complexity overhead that probably doesn't matter much. Overall, I prefer avoiding the perf overhead, but it's not a strong opinion.

Reply:
@dzhulgakov the main motivation for this is just that in general, we want to minimize boxed/unboxed transitions as we move down the redispatch chain. This situation here is novel because the profiling logic is introducing a unilateral boxing step - i.e. one that doesn't arise strictly from how an op's kernels were implemented - so it's the first time we're in the position of having both forms available. Like @smessmer I don't think it's critical, but my reasons for preferring it this way are basically
* It's not impossible that we might later want to do a similar thing in other contexts (i.e., internal dispatcher logic provokes a boxing - or more significantly, unboxing - step that we want to reuse, or possibly that we add it as part of the redispatching API)
* I think we should treat the dispatcher as a big bundle of hot paths and err on the side of conserving work when it can be done without making the logic too ugly.  I think this clears that bar, but wdyt?

Reply:
I don't have strong opinion either. I just don't feel shaving off extra boxing from profiling codepath is that critical perf wise. However, since it's already implemented, we can keep it. As you said, it's good to have all tools in the toolbox